### PR TITLE
Add az auth to postgres flexible

### PR DESCRIPTION
**Pull request by isevenbe**
# [1800](https://github.com/aztfmod/terraform-azurerm-caf/issues/1800)

## PR Checklist

---

- [X] I have added example(s) inside the [./examples/postgresql_flexible_server/103-simple-postgresql-flexible-with-az-auth/configuration.tfvars] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [X] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [X] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Adding the support of authentication block inside the azurerm_postgresql_flexible_server module ref: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server#authentication

## Does this introduce a breaking change

- [ ] YES
- [X] NO

### CODE REVIEW

File: modules/databases/postgresql_flexible_server/server.tf [hunk 1]
```diff
@@ -27,6 +27,12 @@ resource "azurerm_postgresql_flexible_server" "postgresql" {
   administrator_login    = try(var.settings.create_mode, "Default") == "Default" ? try(var.settings.administrator_username, "pgadmin") : null
   administrator_password = try(var.settings.create_mode, "Default") == "Default" ? try(var.settings.administrator_password, azurerm_key_vault_secret.postgresql_administrator_password.0.value) : null
 
+  authentication {
+    active_directory_auth_enabled = try(var.settings.authentication.active_directory_auth_enabled, false)
```
Comment (arnaudlh) [inline modules/databases/postgresql_flexible_server/server.tf:31]:
```suggestion
    active_directory_auth_enabled = try(var.settings.authentication.active_directory_auth_enabled, null)
```

File: modules/databases/postgresql_flexible_server/server.tf [hunk 2]
```diff
@@ -27,6 +27,12 @@ resource "azurerm_postgresql_flexible_server" "postgresql" {
...
+    password_auth_enabled         = try(var.settings.authentication.password_auth_enabled, true)
```
Comment (arnaudlh) [inline modules/databases/postgresql_flexible_server/server.tf:32]:
```suggestion
    password_auth_enabled         = try(var.settings.authentication.password_auth_enabled, null)
```

File: modules/databases/postgresql_flexible_server/server.tf [hunk 3]
```diff
@@ -27,6 +27,12 @@ resource "azurerm_postgresql_flexible_server" "postgresql" {
...
+    tenant_id                     = try(var.settings.authentication.tenant_id, "")
```
Comment (arnaudlh) [inline modules/databases/postgresql_flexible_server/server.tf:33]: Can we add the current ```tenant_id``` from vars if none is specified?

File: modules/databases/postgresql_flexible_server/server.tf [hunk 4]
```diff
@@ -27,6 +27,12 @@ resource "azurerm_postgresql_flexible_server" "postgresql" {
   administrator_login    = try(var.settings.create_mode, "Default") == "Default" ? try(var.settings.administrator_username, "pgadmin") : null
   administrator_password = try(var.settings.create_mode, "Default") == "Default" ? try(var.settings.administrator_password, azurerm_key_vault_secret.postgresql_administrator_password.0.value) : null
 
+  authentication {
```
Comment (arnaudlh) [inline modules/databases/postgresql_flexible_server/server.tf:30]: should be optional block (see pattern in other parts of the code)

**Review by arnaudlh — CHANGES_REQUESTED** [review]
Please review.

File: modules/databases/postgresql_flexible_server/server.tf (hunk 3 above)
Comment (isevenbe) [reply modules/databases/postgresql_flexible_server/server.tf]: I don't gezt what do you mean, have you an example that could help me to understand ? :)

**Comment by isevenbe** [comment]
@arnaudlh i have reviewed

File: modules/databases/postgresql_flexible_server/server.tf (hunk 3 above)
Comment (arnaudlh) [reply modules/databases/postgresql_flexible_server/server.tf]: Thinking about something like this:

```
tenant_id = can(var.settings.authentication.tenant_id) ? var.settings.authentication.tenant_id : var.client_config.tenant_id
```

And alongside, comment ```tenant_id``` in the example the tenant_id, so it passes CI/CD.

Comment (isevenbe) [reply modules/databases/postgresql_flexible_server/server.tf]: Ok i get it !

**Comment by ALA-DQ** [comment]
Up, I need this feature as well, asap, for my company.

File: examples/postgresql_flexible_server/103-simple-postgresql-flexible-with-az-auth/configuration.tfvars [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+    authentication = {
+      # (Optional) Whether or not Active Directory authentication is allowed to access the PostgreSQL Flexible Server. Defaults to false.
+      active_directory_auth_enabled = true
+
+      # (Optional) Whether or not password authentication is allowed to access the PostgreSQL Flexible Server. Defaults to true.
+      password_auth_enabled = true
+
+      # (Optional) The Tenant ID of the Azure Active Directory which is used by the Active Directory authentication. active_directory_auth_enabled must be set to true.
+      tenant_id = "00000-ee35-4265-95f6-46e9a9b4ec96"
```
Comment (arnaudlh) [inline examples/postgresql_flexible_server/103-simple-postgresql-flexible-with-az-auth/configuration.tfvars:40]:
```suggestion
      #tenant_id = "00000-ee35-4265-95f6-46e9a9b4ec96"
```

**Review by arnaudlh — APPROVED** [review]
LGTM

**Comment by arnaudlh** [comment]
thanks for your contribution @isevenbe!

**Pull request merged**